libyul/backends/evm/ssa/Stack.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/backends/evm/ssa/Stack.h>
+
+#include <fmt/ranges.h>
+
+#include <range/v3/view/drop.hpp>
+
+namespace
+{
+size_t junkTailSize(solidity::yul::ssa::StackData const& _stackData)
+{
+	std::size_t numJunk = 0;
+	auto it = _stackData.begin();
+	while (it != _stackData.end() && it->isJunk())
+	{
+		++numJunk;
+		++it;
+	}
+	return numJunk;
```

Comment:
How about something like this?
```suggestion
auto junkTail = _stackData | ranges::views::take_while([](const auto& slot) { return slot.isJunk(); });
return ranges::distance(junkTail);
```
Or with `find_if_not`:
```
auto it = std::find_if_not(_stackData.begin(), _stackData.end(), [](const auto& slot) { return slot.isJunk(); });
return std::distance(_stackData.begin(), it);
```

Reply:
Ahh that's nice. I wasn't aware that `ranges::distance` existed. Updated the implementation accordingly.